### DISCUSSION TITLE
No Active Filter Set

### DISCUSSION THREAD

**Discussion by mikelebon** [zwave-js/node-red-contrib-zwave-js#263] [Q&A] [answered]
Hello,

I'm new to this contrib and to Z-Wave in general.  I managed to get a basic flow working quite easily and I am very impressed so far!

There is a part of the UI that I don't understand but that I feel would likely be useful if I knew how to use it.

I'm talking about this:

<img width="478" alt="Screen Shot 2022-12-14 at 5 12 16 AM" src="https://user-images.githubusercontent.com/9260044/207568232-5b39ac22-ec89-47dd-b525-2b380b54e531.png">

Why doesn't the "Add To Filter Set" button work?  What does this message mean?  How is this supposed to work?
I cannot find anything related to this functionality in the Wiki.

Here is my flow:

`[{"id":"d8c32580b41909fa","type":"tab","label":"Flow 1","disabled":false,"info":"","env":[]},{"id":"86dde574aadde0f4","type":"junction","z":"d8c32580b41909fa","x":680,"y":280,"wires":[["506d3955fa812c3a"]]},{"id":"d12ff1105c590f91","type":"zwave-js","z":"d8c32580b41909fa","networkIdentifier":"1","serialPort":"/dev/ttyUSB0","name":"ZWave Controller","encryptionKey":"693c7737915ef66e93e0e9608155b47c","encryptionKeyS2U":"0c4903bb90d930ad6c1ccc40bdc39604","encryptionKeyS2A":"7fceae2c1a8b2d3695efe0cbb11979d7","encryptionKeyS2AC":"29592363df039a114a39f9ff89869332","ackTimeout":"","controllerTimeout":"","sendResponseTimeout":"","sendDataCallback":"","serialAPIStarted":"","logLevelPin":"none","logLevel":"none","logFile":"","logNodeFilter":"","sendUsageStatistics":false,"valueCacheDiskThrottle":"normal","customConfigPath":"","intvwUserCodes":false,"softResetUSB":false,"outputs":1,"disableOptimisticValueUpdate":false,"FWlicenseKey":"NON-COMMERCIAL","scalesTemp":"0x00","scalesHumidity":"0x00","x":210,"y":520,"wires":[["aba75d6fe04bcd2a"]]},{"id":"aba75d6fe04bcd2a","type":"debug","z":"d8c32580b41909fa","name":"","active":false,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":410,"y":520,"wires":[]},{"id":"506d3955fa812c3a","type":"zwave-device","z":"d8c32580b41909fa","networkIdentifier":1,"name":"Fibaro Wall Plug","filteredNodeId":"5","multicast":false,"datamode":"Send/Receive","messagesPerMS":1,"messageInterval":250,"isolated":false,"outputs":1,"inputs":1,"showStatus":true,"x":860,"y":280,"wires":[["e7183a8f7db34a52"]]},{"id":"841a02cdcdcedbe1","type":"inject","z":"d8c32580b41909fa","name":"Switch ON","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"{\"mode\":\"ValueAPI\",\"method\":\"setValue\",\"params\":[{\"commandClassName\":\"Binary Switch\",\"commandClass\":37,\"endpoint\":0,\"property\":\"targetValue\",\"propertyName\":\"targetValue\"},true]}","payloadType":"json","x":540,"y":260,"wires":[["86dde574aadde0f4"]]},{"id":"2c52f84950e73fcd","type":"inject","z":"d8c32580b41909fa","name":"Switch OFF","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"{\"mode\":\"ValueAPI\",\"method\":\"setValue\",\"params\":[{\"commandClassName\":\"Binary Switch\",\"commandClass\":37,\"endpoint\":0,\"property\":\"targetValue\",\"propertyName\":\"targetValue\"},false]}","payloadType":"json","x":530,"y":300,"wires":[["86dde574aadde0f4"]]},{"id":"82d0fd448444c8c5","type":"zwave-device","z":"d8c32580b41909fa","networkIdentifier":1,"name":"Aeotec MultiSensor 7","filteredNodeId":"3","multicast":false,"datamode":"Send/Receive","messagesPerMS":1,"messageInterval":250,"isolated":false,"outputs":1,"inputs":1,"showStatus":true,"x":200,"y":220,"wires":[["f2307f4c04b24081","f79b412fa5c6ac7e","a7c1ed72a998e616"]]},{"id":"e7183a8f7db34a52","type":"debug","z":"d8c32580b41909fa","name":"Wall Plug","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":1080,"y":280,"wires":[]},{"id":"f2307f4c04b24081","type":"debug","z":"d8c32580b41909fa","name":"MultiSensor 7","active":false,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":500,"y":100,"wires":[]},{"id":"f79b412fa5c6ac7e","type":"function","z":"d8c32580b41909fa","name":"Trigger on Motion","func":"if (msg.payload.object?.propertyName !== \"Motion\") return;\n\nconst motionDetected = msg.payload.object.normalizedObject.newValue;\n\nconst payload = {\n    \"mode\": \"ValueAPI\",\n    \"method\": \"setValue\",\n    \"params\": [\n        {\n            \"commandClassName\": \"Binary Switch\",\n            \"commandClass\": 37,\n            \"endpoint\": 0,\n            \"property\": \"targetValue\",\n            \"propertyName\": \"targetValue\"\n        },\n        motionDetected\n    ]\n};\n\n\nreturn { payload };","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":510,"y":220,"wires":[["1e9f0a848f3cdd21","86dde574aadde0f4"]]},{"id":"1e9f0a848f3cdd21","type":"debug","z":"d8c32580b41909fa","name":"Motion Command","active":false,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":770,"y":220,"wires":[]},{"id":"d3757b0b1e38cb33","type":"zwave-device","z":"d8c32580b41909fa","networkIdentifier":1,"name":"Aeotec Door Sensor","filteredNodeId":"6","multicast":false,"datamode":"Send/Receive","messagesPerMS":1,"messageInterval":250,"isolated":false,"outputs":1,"inputs":1,"showStatus":true,"x":200,"y":340,"wires":[["081ac80337bbd35c","b52d30d29be498db"]]},{"id":"dafae43bf0589d17","type":"debug","z":"d8c32580b41909fa","name":"Opening Command","active":false,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":770,"y":340,"wires":[]},{"id":"081ac80337bbd35c","type":"function","z":"d8c32580b41909fa","name":"Trigger on Open","func":"if (msg.payload.object?.propertyName !== \"Access Control\") return;\n\nconst sensorOpen = msg.payload.object.newValue === 22;\n\nconst payload = {\n    \"mode\": \"ValueAPI\",\n    \"method\": \"setValue\",\n    \"params\": [\n        {\n            \"commandClassName\": \"Binary Switch\",\n            \"commandClass\": 37,\n            \"endpoint\": 0,\n            \"property\": \"targetValue\",\n            \"propertyName\": \"targetValue\"\n        },\n        sensorOpen\n    ]\n};\n\n\nreturn { payload };","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":520,"y":340,"wires":[["dafae43bf0589d17","86dde574aadde0f4"]]},{"id":"b52d30d29be498db","type":"debug","z":"d8c32580b41909fa","name":"Door Sensor","active":false,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":430,"y":380,"wires":[]},{"id":"a7c1ed72a998e616","type":"event-filter","z":"d8c32580b41909fa","name":"ZWave Event Filter","filters":[{"index":0,"name":"Filter Name","valueIds":[],"events":["VALUE_UPDATED"],"strict":false,"id":"157100"}],"outputs":1,"changeDate":"2022-12-14T09:04:02.793Z","showStatus":true,"x":510,"y":160,"wires":[["b596f6c8c662a503"]]},{"id":"b596f6c8c662a503","type":"debug","z":"d8c32580b41909fa","name":"","active":false,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":690,"y":160,"wires":[]}]`

Thank you for any help!

Mike

**Comment by marcus-j-davies** [comment, marked as answer]
Hi @mikelebon 

`Add To Filter Set` is for when you're  adding a filter and wish to filter to a certain property(s)
as an example, you may wish to split up events

See video below....
`Add To Filter Set` will apply the property filter to the active filter rule 
multiple properties can be used in each rule (its not limited to 1 property)


https://user-images.githubusercontent.com/55892693/207572635-f026ceea-fc90-42a1-8cab-1ca8b4a28d70.mov

**Reply by mikelebon** [reply]
Hello @marcus-j-davies!

That was quick!  Thank you very much :)
This seems to work as expected.

**Reply by marcus-j-davies** [reply]
Anytime!
